### PR TITLE
Update Write Your First Application Tutorial

### PR DESCRIPTION
**Pull request by denali49**
Signed-off-by: Chris Gabriel <chris_gabriel_98@yahoo.com>

#### Type of change

- Documentation update

#### Description

Update Write Your First Application tutorial to the use the new asset-transfer-basic sample.

#### Additional details

This update removes reference to FabCar, and is a significant update based on the new application and chaincode
in fabric-samples.

### CODE REVIEW

File: docs/source/write_first_app.rst [hunk 1]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
+            const contract = network.getContract(myChaincodeName);
 
-.. code:: bash
+Fourth, initializing the ledger with some data we can work with
+---------------------------------------------------------------
 
-  const gateway = new Gateway();
-  await gateway.connect(ccpPath, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+Now that we are to the point where we are actually having the sample application submit transactions, let’s
+go through them in sequence where we'll show the snippet of the application code, the snippet of the corresponding
```
Comment (lehors) [inline docs/source/write_first_app.rst]: where we'll show the snippet of the application code, the snippet of the corresponding -> , look at the snippets of the application code and the corresponding

File: docs/source/write_first_app.rst [hunk 2]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
+            // Get the contract from the network.
+            const contract = network.getContract(myChaincodeName);
 
-.. code:: bash
+Fourth, initializing the ledger with some data we can work with
+---------------------------------------------------------------
 
-  const gateway = new Gateway();
-  await gateway.connect(ccpPath, { wallet, identity: 'appUser', discovery: { enabled: true, asLocalhost: true } });
+Now that we are to the point where we are actually having the sample application submit transactions, let’s
```
Comment (lehors) [inline docs/source/write_first_app.rst]: to the point  -> at the point

File: docs/source/write_first_app.rst [hunk 3]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
+  Submit Transaction: InitLedger creates the initial set of assets on the ledger.
+
+Fifth, the rest of the transaction submission sequence from the sample application
+----------------------------------------------------------------------------------
+
+First, a word about querying the ledger.
 
-.. code:: bash
+Each peer in a blockchain network hosts a copy of the `ledger <./ledger/ledger.html>`_. An application
+program can view the most recent data from the ledger using read only invocations of
```
Comment (lehors) [inline docs/source/write_first_app.rst]: read only -> read-only

File: docs/source/write_first_app.rst [hunk 4]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
+        console.log('\nSubmit Transaction: UpdateAsset asset70');
+        //Non existing asset asset70 should throw Error
+        await contract.submitTransaction('UpdateAsset', 'asset70', 'blue', '5', 'Tomoko', '300');
+    } catch (error) {
+        console.log(`Expected an error on UpdateAsset of non-existing Asset: ${error}`);
+    }
 
-Let's take a closer look at the ``queryAllCars`` transaction to see how it
-interacts with the ledger.
+Chaincode -- ``'UdpateAsset'``
```
Comment (lehors) [inline docs/source/write_first_app.rst]: UpdateAsset

File: docs/source/write_first_app.rst [hunk 5]
```diff
@@ -512,100 +880,42 @@ the application, smart contract, peers and ordering service work together to
...
-.. code:: bash
-
-   Wallet path: ...fabric-samples/fabcar/javascript/wallet
-   Transaction has been evaluated, result is:
-   {"color":"Black","docType":"car","make":"Honda","model":"Accord","owner":"Dave"}
+The ``evaluateTransaction`` method represents one of the simplest interactions
+with a smart contract in blockchain network. It simply picks a peer defined in
+the connection profile and sends the request to it, where it is evaluated. The
+smart contract queries all the cars on the peer's copy of the ledger and returns
+the result to the application. This interaction does not result in an update the
```
Comment (lehors) [inline docs/source/write_first_app.rst]: and update the -> an update of the

File: docs/source/write_first_app.rst [hunk 6]
```diff
@@ -512,100 +880,42 @@ the application, smart contract, peers and ordering service work together to
...
+ledger.
 
-The ownership of ``CAR12`` has been changed from Tom to Dave.
+The asset-transfer ('basic') smart contract
+-------------------------
+The smart contract sample is available in following languages:
```
Comment (lehors) [inline docs/source/write_first_app.rst]: in following languages -> in the following languages

**Review by lehors — DISMISSED** [review]
LGTM overall (I tested it) but I found a few typos and made a few suggestions for improvement

File: docs/source/write_first_app.rst [hunk 7]
```diff
@@ -43,13 +53,14 @@ We’ll go through three principle steps:
   .. image:: images/AppConceptsOverview.png
 
   **2. Explore a sample smart contract.**
-  We’ll inspect the sample Fabcar smart contract to learn about the transactions within them,
+  We’ll inspect the sample assetTransfer (javascript) smart contract to learn about the transactions within them,
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: Done, thanks for the review and comments.

File: docs/source/write_first_app.rst [hunk 8]
```diff
@@ -43,13 +53,14 @@ We’ll go through three principle steps:
...
   and how they are used by applications to query and update the ledger.
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: done.

File: docs/source/write_first_app.rst [hunk 9]
```diff
@@ -43,13 +53,14 @@ We’ll go through three principle steps:
...
 
   **3. Interact with the smart contract with a sample application.** Our application will
-  use the FabCar smart contract to query and update car assets on the ledger.
-  We'll get into the code of the apps and the transactions they create,
-  including querying a car, querying a range of cars, and creating a new car.
+  use the assetTransfer smart contract to create, query, and update assets on the ledger.
+  We'll get into the code of the apps and the transactions they create, including initializing
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: done.

File: docs/source/write_first_app.rst [hunk 10]
```diff
@@ -90,55 +101,68 @@ bringing up a new one.
...
+
+Launch the Fabric test network using the ``network.sh`` shell script.
+
+.. code:: bash
+
+  ./network.sh up createChannel -c mychannel -ca
+
+This command will deploy the Fabric test network with two peers, three certificate authorities (Orderer, Org1, Org2),
+and an ordering service. Instead of using the cryptogen tool, we bring up the test network using Certificate Authorities, 
+hence the ``-ca`` flag. Additionally, the org admin user registration was bootstrapped when the Certificate Authority was started.
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: done.

File: docs/source/write_first_app.rst (hunk 1 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst (hunk 2 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst (hunk 3 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst (hunk 4 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst (hunk 5 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst (hunk 6 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst [hunk 11]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
-Updating the ledger
--------------------
+.. code:: bash
 
-Now that we’ve done a few ledger queries and added a bit of code, we’re ready to
-update the ledger. There are a lot of potential updates we could make, but
-let's start by creating a **new** car.
+  const identity = await wallet.get(registerUser.ApplicationUserId);
 
-From an application perspective, updating the ledger is simple. An application
```
Comment (sijocherian) [inline docs/source/write_first_app.rst:441]: I think this para about Update to the ledger got lost in the new description. Seem like a good detail to keep along with the tutorial/write_first_app.diagram.2.png diagram

File: docs/source/write_first_app.rst [hunk 12]
```diff
@@ -512,100 +880,42 @@ the application, smart contract, peers and ordering service work together to
...
-  node query.js
-
-It should return this result:
-
-.. code:: bash
-
-   Wallet path: ...fabric-samples/fabcar/javascript/wallet
-   Transaction has been evaluated, result is:
-   {"color":"Black","docType":"car","make":"Honda","model":"Accord","owner":"Dave"}
+The ``evaluateTransaction`` method represents one of the simplest interactions
```
Comment (sijocherian) [inline docs/source/write_first_app.rst]: if possible, move this para about evaluateTransaction before the description about `submitTransaction` in line 859, to describe the simpler evaluateTransaction first as a natural progression

**Review by sijocherian — CHANGES_REQUESTED** [review]
Great to see the application & chaincode description side by side

File: docs/source/write_first_app.rst [hunk 13]
```diff
@@ -90,55 +101,68 @@ bringing up a new one.
...
-Launch your network using the ``startFabric.sh`` shell script.
+Next, let's deploy the chaincode by calling the ``./network.sh`` script with the chaincode name and language options.
 
 .. code:: bash
 
-  ./startFabric.sh javascript
+  ./network.sh deployCC -ccn basic -ccl javascript
+
+.. note:: Behind the scenes, this script uses the chaincode lifecycle to package, install,
+          query installed, approve for my org, and commit chaincode.
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: Thank you for your review, done.

File: docs/source/write_first_app.rst [hunk 14]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
+
+Third, how the sample application interacts with the chaincode
+-----------------------------------------------------------
+
+In the prior steps, the application generated the admin and app user credentials and placed them in the wallet.
+If the credentials exist and have the correct permissions attributes associated with them, the sample application user
+will be able to call chaincode functionality from the sample application after getting reference to the channel name and contract name.
+
+You will notice that in the following lines of application code, the application is getting reference
+to the channel on the network and the contract name:
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst]: Done.

File: docs/source/write_first_app.rst [hunk 15]
```diff
@@ -164,329 +188,673 @@ You should see the following:
...
```
[the remark this answers is not in the document]
Comment (denali49) [reply docs/source/write_first_app.rst:320]: Thanks for the reminder and the suggested text!  Done.

File: docs/source/write_first_app.rst (hunk 11 above)
Comment (denali49) [reply docs/source/write_first_app.rst:441]: Agreed, re-added paragraph and image.

File: docs/source/write_first_app.rst (hunk 12 above)
Comment (denali49) [reply docs/source/write_first_app.rst]: Agree, it flows better this way.  Done.

**Review by sijocherian — APPROVED** [review]

**Review by lehors — APPROVED** [review]
LGTM. Thanks.

**Review by denyeart — APPROVED** [review]
Thanks Chris!

**Comment by denyeart** [comment]
Some additional tutorial updates in https://github.com/hyperledger/fabric/pull/1687

**Comment by denali49** [comment]
@Mergifyio backport release-2.2

**Pull request merged**